sympy/matrices/common.py

```diff
@@ -2218,37 +2218,42 @@ def __neg__(self):
         return self._eval_scalar_mul(-1)
 
     @call_highest_priority('__rpow__')
-    def __pow__(self, num):
+    def __pow__(self, exp):
         if self.rows != self.cols:
             raise NonSquareMatrixError()
         a = self
-        jordan_pow = getattr(a, '_matrix_pow_by_jordan_blocks', None)
-        num = sympify(num)
-        if num.is_Number and num % 1 == 0:
+        exp = sympify(exp)
+        jordan_pow = getattr (a, '_matrix_pow_by_jordan_blocks', None)
+
+        if exp.is_Number and exp % 1 == 0:
```

Comment:
This seems like a cryptic way of checking for an integer. I guess it's supposed to work for Float or Integer...

Reply:
Its what was there, I am trying to keep changes to minimum necessary to make it work well since every change made has potential effects outside of what I am aware...